### DISCUSSION TITLE
Need Assistance - Unable to Run and Access OpenCRVS

### DISCUSSION THREAD

**Discussion by tntra-rajdeep** [opencrvs/opencrvs-core#3610] [Technical Q&A] [answered]
Hi Team,

We installed OpenCRVS Core locally from your documentation. We are able to install and login into the OpenCRVS portal (Port:3020) However, After login successfully its redirects to the following page and shows the error. Please find the attached screenshot for your reference. 

![Pasted Image_ Jun 30, 2022 - 5_36_16pm](https://user-images.githubusercontent.com/105046784/176700279-ceb99f4a-f94d-4689-b66c-c1833ed41fa9.png) 

For OpenHIM admin console, we are not able to login in due to a change of default password( as mentioned in your document). Please update your document with an updated password or share it with us. 

Please let me know if you have any queries or concerns at rajdeep.randhava@tntra.io. 

Regards,
Rajdeep

**Comment by rikukissa** [comment]
Hey Rajdeep,

Great to see you're testing OpenCRVS, and I'm sorry you're encountering difficulties. To help me reproduce the issue locally, could I please ask you to provide me with the following information:

1. User you logged in with (I'm assuming jonathan.campbell, but I want to make sure)
2. The network response you're getting from the local server when the application fetches the performance data. 

You can find this by first opening up Developer Tools:
<img width="743" alt="image" src="https://user-images.githubusercontent.com/1206987/176842618-05740e81-a66e-4fba-a970-aa3c782a99d8.png">

And navigation onto the **Network** - tab. Reload the page with the tab open if you do not see any requests listed here. From the list of requests, you should find one labelled as `graphql` where on the Payload tab, the **operationName** key has a value of `getTotalMetrics`.

<img width="709" alt="image" src="https://user-images.githubusercontent.com/1206987/176843100-6e271584-0012-4fdf-aabf-b1d51bfbde4a.png">

Navigate onto the "Response" tab and send me a screenshot/copy the value here, please. 

Best regards,
Riku Rouvila

**Reply by shukladhaval25** [reply]
> 
![Screenshot from 2022-07-01 13-13-27](https://user-images.githubusercontent.com/17428629/176857737-ddaf4bb5-6d49-4ad3-a01e-f4f669cb78c1.png)
(1) we are using Jonathan user id and password as per document details.

**Reply by rikukissa** [reply]
Thanks! Could you also send me a screenshot of the "Preview" tab, with the data expanded, please?

**Reply by shukladhaval25** [reply]
![Screenshot from 2022-07-01 15-32-53](https://user-images.githubusercontent.com/17428629/176873611-511f8261-b783-40db-aee2-7ced634b99f0.png)

**Reply by rikukissa** [reply]
Interesting.. It seems like you're `gateway` service would be down. Are you getting any response if you navigate to http://localhost:7070/ping?service=metrics 

This might be because the port 7070 was in use when you started the stack. Restarting it might help and I would keep track of the logs, following lines with `@opencrvs/gateway:` prefix. Anything that resembles an error would help.

You can check the status of each service by using these URLs:

http://localhost:7070/ping?service=auth
http://localhost:7070/ping?service=user-mgnt
http://localhost:7070/ping?service=metrics
http://localhost:7070/ping?service=notification
http://localhost:7070/ping?service=countryconfig
http://localhost:7070/ping?service=search
http://localhost:7070/ping?service=workflow
http://localhost:7070/ping?service=gateway

They should all return a response like this:

<img width="1275" alt="image" src="https://user-images.githubusercontent.com/1206987/176875751-57b3e7ae-5db9-4ba9-9487-b4a9c3e058fa.png">

**Reply by shukladhaval25** [reply]
While checking with http://localhost:7070/ping?service=metrics, we found that localhost:7070 is not running. it's not giving any response back. What we have to do to start service on localhost:7070? Day before yesterday it was running but after that it's constantly fail. 

![image](https://user-images.githubusercontent.com/17428629/177483018-a45cc31e-ef21-4044-897b-6eb7b2935331.png)

**Reply by shukladhaval25** [reply]
Get suck and not moving ahead. 

![image](https://user-images.githubusercontent.com/17428629/177485016-3b8400e4-afc7-489e-b5eb-0ed7fd95b414.png)

**Reply by Zangetsu101** [reply]
Hi @shukladhaval25,
It seems that `gateway` microservice (which runs on port 7070) is not starting up that's why you are getting the error. It could be because some other process is using the port 7070 that's why it's not starting up.
Could you please run the command `sudo lsof -i -P -n | grep LISTEN` before starting the project and see if the port is already in use or not?

**Reply by Zangetsu101** [reply]
If so then you can use `sudo fuser -k 7070/tcp` to kill that process then try running the project and the gateway should be able to start properly.

**Comment by shukladhaval25** [comment, marked as answer]
Thanks for instance reply.  Now it's working and running fine. Once again thanks for quick response. 
Thanking you,Dhaval  shukla
    On Wednesday, 6 July 2022 at 12:49:29 GMT+5:30, Tameem Bin Haider ***@***.***> wrote:  
 
 


If so then you can use sudo fuser -k 7070/tcp to kill that process then try running the project and the gateway should be able to start properly.

**Comment by shukladhaval25** [comment]
Not able to login into openCRVS - Admin Console (openHIM)

As per document, I am trying to login into openCRVS - Admin Console screen and I got below error

![image](https://user-images.githubusercontent.com/17428629/177566594-d97996f1-d597-47c6-bdc1-e56ea1781f96.png)

I have follow the steps given in document as below but not able to login based on user name and password mention in document.
![image](https://user-images.githubusercontent.com/17428629/177566996-6fc37cde-d482-4a36-94f2-0f54f30b6823.png)

I get stuck over here. My openCRVS on port 3020 is working fine but not able to login in openHIM 8080 port.

**Reply by shukladhaval25** [reply]
@rikukissa 
Do you have any suggestion for login issue with openHIM which I posted up previously?

**Reply by Zangetsu101** [reply]
@shukladhaval25 Sorry about the confusion. Actually we updated the password for OpenHIM but haven't updated the docs yet. The new password is `wXV8xSW2Ju5X3EPn`.

**Reply by shukladhaval25** [reply]
Thanks for you support it's working now.

**Comment by euanmillar** [comment]
@Zangetsu101 @rikukissa  I suggest that we use the same approach [here](https://github.com/opencrvs/opencrvs-core/blob/5550f2454d8af080c0b4292de16735396ddaf83c/setup.sh#L270) in the **bash setup.sh** command to check running ports and stop them in the **yarn dev** command to help other users who may have this problem in future.
